8thwall-camera.js

```diff
@@ -5,88 +5,402 @@
  * Use this for SLAM tracking based on 8thwall.
  *
  * Make sure to enable 8thwall in "Project Settings" > "AR". See also the
- * [AR Getting Started Guide](/getting-started/quick-start-ar). */
+ * [AR Getting Started Guide](/getting-started/quick-start-ar)
+ * 
+ * [WIP] 
+ * New 8thwall camera component
+ * Currently only works with BACK camera.
+ *
+ * - remove any occurrences of the old 8thwall-camera component in the editor
+ * - Add this component to NonVrCamera
+ * 
+ * IMPORTANT!
+ * - until it get's fixed, select 'customIndexHtml' in the project settings
+ * - open a generated index.html and crossorigin="anonymous" attribute to the <script> tag where //apps.8thwall.com/xrweb is loaded
+ * 
+ * 
+ * ######### Using UI overlays #############
+ * By default, this component renders own UI to give the user feedback about rejected permissions or tracking errors.
+ * It can be changed by enabling useCustomUIOverlays flag.
+ * 
+ * if useCustomUIOverlays is enabled, you are expected handle the following events dispatched by the window object:
+ * - "8thwall-request-user-interaction" - used only on iOS safari. Request a user to perform and interaction with the page so that javascript is allowed to allowed to request a motion/camera/mic permissions.
+ *  Make sure the window object is dispatching an '8thwall-permissions-allowed' event after interaction has happened. 
+ * ```
+ * <button onclick="window.dispatchEvent(new Event('8thwall-permissions-allowed'))">Allow Sensors</button>
+ * ```
+ * 
+ * - "8thwall-permission-fail" - user rejected any of the permissions
+ * 
+ * - "8thwall-error" - runtime 8thwall error occurred
```

Comment:
```suggestion
 * - "8thwall-request-user-interaction" - used only on iOS Safari. Request a user to perform and interaction with the
 * page so that JavaScript is allowed to use device motion, camera and microphone.
 * Make sure the window object dispatches an '8thwall-permissions-allowed' event after interaction has happened. 
 * ```html
 * <button onclick="window.dispatchEvent(new Event('8thwall-permissions-allowed'))">Allow Sensors</button>
 * ```
 * - "8thwall-permission-fail" - user rejected any of the permissions.
 * - "8thwall-error" - runtime 8thwall error occurred.
```

Would that onclick work, though? That seems to be missing the actual request for permissions. Maybe that can also be outlined for the user here.